src/api/httpClient.ts

```diff
@@ -0,0 +1,25 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+
+export interface HttpClient {
+  get<T = any>(url: string, config?: AxiosRequestConfig): Promise<AxiosResponse<T>>;
+  post<T = any>(url: string, data?: any, config?: AxiosRequestConfig): Promise<AxiosResponse<T>>;
+  put<T = any>(url: string, data?: any, config?: AxiosRequestConfig): Promise<AxiosResponse<T>>;
+  patch<T = any>(url: string, data?: any, config?: AxiosRequestConfig): Promise<AxiosResponse<T>>;
+  delete<T = any>(url: string, config?: AxiosRequestConfig): Promise<AxiosResponse<T>>;
+}
```

Comment:
Podia ter simplificado pra não fazer isso tudo na mão:
```suggestion
export type HttpClient = {
  get: typeof axios.get
  post: typeof axios.post
  put: typeof axios.put
  patch: typeof axios.patch
}

```